prow/jira/jira.go

```diff
@@ -190,6 +191,23 @@ func (jc *client) AddRemoteLink(id string, link *jira.RemoteLink) error {
 	return nil
 }
 
+func (jc *client) UpdateRemoteLink(id string, link *jira.RemoteLink) error {
+	internalLinkId := fmt.Sprint(link.ID)
+	req, err := jc.upstream.NewRequest("PUT", "rest/api/2/issue/"+id+"/remotelink/"+internalLinkId, link)
+	if err != nil {
+		return fmt.Errorf("failed to construct request: %w", err)
+	}
+	resp, err := jc.upstream.Do(req, nil)
+	if resp != nil {
+		defer resp.Body.Close()
+	}
+	if err != nil {
+		return fmt.Errorf("failed to update link: %w", JiraError(resp, err))
+	}
```

Comment:
Please also check the returned status code, if e.G. statuscode is 503 but you got a valid response, err will be nil

Reply:
I have allowed 204 responses only as documentation stated it is the only status code to be considered "valid"